zml/module.zig

```diff
@@ -565,47 +644,57 @@ fn assignBlockArguments(v: anytype, block: mlir.Block, start: usize) usize {
 }
 
 /// Visit the given struct and fill the `buffers` slice with the buffer associated with encountered Tensor.
-fn fillBuffers(v: anytype, buffers: []*pjrt.Buffer) void {
+fn fillBuffers(v: anytype, buffers: []const [*]*pjrt.Buffer, start: u32, len: u32) void {
     const LocalContext = struct {
-        index: usize,
-        buffers: []*pjrt.Buffer,
+        index: u32,
+        buffers: []const [*]*pjrt.Buffer,
     };
-    var ctx: LocalContext = .{
-        .index = 0,
+    var capture: LocalContext = .{
+        .index = start,
```

Comment:
We should try to be consistent on the name of the `LocalContext` variable

Reply:
agree, but I'm not a big fan of `local_context` and `inner_context`, was wondering if something like `capture` would make it easier to follow to people coming from languages where "capture" is a thing.

Reply:
(renamed to context)

Reply:
I've seen the `inner_` prefix used many times, that's why I originally used it. But I get your point